tc_solaris.go

```diff
@@ -0,0 +1,75 @@
+package console
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+func tcget(fd uintptr, p *unix.Termios) error {
+	var err error
+	p, err = unix.IoctlGetTermios(int(fd), unix.TCGETS)
+	return err
+}
+
+func tcset(fd uintptr, p *unix.Termios) error {
+	return unix.IoctlSetTermios(int(fd), unix.TCSETS, p)
+}
+
+func tcgwinsz(fd uintptr) (WinSize, error) {
+	var ws WinSize
+
+	uws, err := unix.IoctlGetWinsize(int(fd), unix.TIOCGWINSZ)
```

Comment:
Ioctl[GS]etWinsize were introduced on linux 11 days ago:

https://github.com/golang/sys/commit/cd2c276457edda6df7fb04895d3fd6a6add42926

so i don't think we can commonize this until after the next version of go is released and we want to make it a build requirement.